Camera type mismatch between raylib and sokol backends in Game.getCamera()

## Description

The `Game.getCamera()` method returns a camera type that is parameterized by the backend, but the return type annotation uses a hardcoded backend type, causing type mismatches when switching backends.

## Error

```
/Users/alexandrecalvao/prj/labelle-toolkit/labelle-engine/engine/game.zig:632:50: error: expected type '*camera.camera.CameraWith(backend.backend.Backend(backend.raylib_backend.RaylibBackend))', found '*camera.camera.CameraWith(backend.backend.Backend(backend.sokol_backend.SokolBackend))'
            return self.retained_engine.getCamera();
```

## Location

`engine/game.zig:631-632`

```zig
pub fn getCamera(self: *Self) *labelle.Camera {
    return self.retained_engine.getCamera();
}
```

## Root Cause

The return type `*labelle.Camera` appears to resolve to a specific backend's camera type at compile time, rather than being backend-agnostic.

## Expected Behavior

`getCamera()` should return the correct camera type for whichever backend is selected at build time.

## Environment

- Building with `-Dbackend=sokol`
- Target: `aarch64-ios-simulator`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/engine/game.zig
+++ b/engine/game.zig
@@ -636,11 +636,13 @@ pub fn GameWith(comptime Hooks: type) type {
         // ==================== Multi-Camera ====================
 
         /// Get a camera by index (0-3)
+        /// Returns the backend-aware camera type from RetainedEngine
         pub fn getCameraAt(self: *Self, index: u2) *RetainedEngine.CameraType {
             return self.retained_engine.getCameraAt(index);
         }
 
         /// Get the camera manager (for advanced multi-camera control)
+        /// Returns the backend-aware camera manager type from RetainedEngine
         pub fn getCameraManager(self: *Self) *RetainedEngine.CameraManagerType {
             return self.retained_engine.getCameraManager();
         }

